quilt/test/test_install.py

```diff
@@ -108,6 +117,51 @@ def test_bad_object_hash(self):
 
         assert not os.path.exists('quilt_packages/foo/bar.json')
 
+    def test_resume_download(self):
+        """
+        Test that existing objects don't get re-downloaded - unless their hash is wrong.
+        """
+        file_data_list = []
+        file_hash_list = []
+        for i in range(3):
+            file_data = "file%d" % i
+            h = hashlib.new(HASH_TYPE)
+            h.update(file_data.encode('utf-8'))
+            file_data_list.append(file_data)
+            file_hash_list.append(h.hexdigest())
+
+        contents = RootNode(dict(
+            file0=FileNode([file_hash_list[0]]),
+            file1=FileNode([file_hash_list[1]]),
+            file2=FileNode([file_hash_list[2]]),
+        ), format=PackageFormat.HDF5)
+        contents_hash = hash_contents(contents)
+
+        os.makedirs('quilt_packages/objs')
+
+        # file0 already exists.
+        with open('quilt_packages/objs/{hash}'.format(hash=file_hash_list[0]), 'w') as fd:
+            fd.write(file_data_list[0])
+
+        # file1 exists, but has the wrong contents.
+        with open('quilt_packages/objs/{hash}'.format(hash=file_hash_list[1]), 'w') as fd:
+            fd.write("Garbage")
+
+        # file2 does not exist.
```

Comment:
That's the usual case that's already covered by other tests, so I'm not doing anything to test it here. The point of this test is to make sure things don't get downloaded unnecessarily.